File: fastapi/app/schemas/nebula.py
from pydantic import BaseModel
from datetime import datetime
from enum import Enum
from typing import Optional

class NebulaStatus(str, Enum):
    ACTIVE = "ACTIVE"
    COMPLETED = "COMPLETED"
    PAUSED = "PAUSED"
    ABANDONED = "ABANDONED"

class NebulaBase(BaseModel):
    title: str
    description: Optional[str] = None
    color_hex: str
    target_date: Optional[datetime] = None
    status: NebulaStatus
    progress: float = 0.0

class NebulaCreate(NebulaBase):
    pass

class NebulaUpdate(NebulaBase):
    pass

class Nebula(NebulaBase):
    id: str
    created_at: datetime
    updated_at: datetime

    class Config:
        orm_mode = True

File: app/src/main/java/com/abdessamad/orbyt/data/models/Nebula.kt
package com.abdessamad.orbyt.data.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.Date
import java.util.UUID

enum class NebulaStatus {
    ACTIVE,
    COMPLETED,
    ARCHIVED
}

@Entity(tableName = "nebulas")
data class Nebula(
    @PrimaryKey
    val id: UUID = UUID.randomUUID(),
    val title: String,
    val description: String?,
    val colorHex: String,
    val targetDate: Date?,
    val status: NebulaStatus,
    val progress: Float = 0.0f,
    val createdAt: Date = Date(),
    val updatedAt: Date = Date()
)